codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/Instantiator.kt

```diff
@@ -165,7 +177,9 @@ open class Instantiator(
             writer.conditionalBlock(
                 "Some(",
                 ")",
-                conditional = model.expectShape(memberShape.container) is StructureShape || symbol.isOptional(),
+                // The conditions are not commutative: note client builders always take in `Option<T>`.
```

Comment:
> note client builders always take in `Option<T>`

This might be changing soon with IDLv2 nullability support. Although, if it's calling the `set_*` builder method, those will likely stay the same since someone should be able to unset a field on the builder.

Reply:
It's calling the `set_*` methods; that is existing behavior I'm not changing.